### PR TITLE
Implement data processing in models

### PR DESCRIPTION
**Pull request by benshidanny11**
### In this milestone, I created custom methods:
-  Users
    - A method that returns the 3 most recent posts for a given user.
-  Posts
    - A method that updates the posts counter for a user.
    - A method that returns the 5 most recent comments for a given post.
-  Comments
    - A method that updates the comments counter for a post.
-  Likes
    - A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,9 +1,9 @@
 class User < ApplicationRecord
-  has_many :posts, foreign_key: 'author_id', dependent: :destroy
-  has_many :comments, foreign_key: 'author_id', dependent: :destroy
-  has_many :likes, foreign_key: 'author_id', dependent: :destroy
+  has_many :likes
+  has_many :posts
+  has_many :comments
```
Comment (AnselemOdims) [inline app/models/user.rb]: - [ ] Please kindly include a foreign key that matches the ERD diagram which is `author_id` in this case.

**Hint**
`has_many :posts, foreign_key: 'author_id'`

File: app/models/post.rb [hunk 2]
```diff
@@ -1,15 +1,17 @@
 class Post < ApplicationRecord
-  belongs_to :author, class_name: 'User', counter_cache: :posts_counter
-  has_many :likes, dependent: :destroy
-  has_many :comments, dependent: :destroy
+  belongs_to :user
```
Comment (AnselemOdims) [inline app/models/post.rb]: - [ ] Please kindly change the belongs to relationship so it belongs to author matching the ERC Diagram

**Hint**
`belongs_to :author, class_name: 'User'`

File: app/models/like.rb [hunk 3]
```diff
@@ -1,12 +1,12 @@
 class Like < ApplicationRecord
-  belongs_to :post, counter_cache: :likes_counter
-  belongs_to :author, class_name: 'User'
-
-  after_save :update_likes_counter
+  belongs_to :user
```
Comment (AnselemOdims) [inline app/models/like.rb]: - [ ] Kindly reference the suggestion above

File: app/models/comment.rb [hunk 4]
```diff
@@ -1,12 +1,12 @@
 class Comment < ApplicationRecord
-  belongs_to :author, class_name: 'User'
-  belongs_to :post, counter_cache: :comments_counter
-
-  after_save :update_comments_counter
+  belongs_to :user
```
Comment (AnselemOdims) [inline app/models/comment.rb]: - [ ] Kindly reference the suggestion above

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @benshidanny11 , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by KayLemba — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hello  @benshidanny11 , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

HIGHLIGHTS:

- Your project is professional 
- Your repo is professional 
- No Linter Errors




Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**